### PR TITLE
Max global steps

### PR DESCRIPTION
**Pull request by priyakasimbeg**
Add flag for max_global_steps.

### CODE REVIEW

File: submission_runner.py [hunk 1]
```diff
@@ -300,6 +301,8 @@ def train_once(
     except spec.TrainingCompleteError:
       train_state['training_complete'] = True
     global_step += 1
+    if global_step == max_global_steps:
```
Comment (znado) [inline submission_runner.py:304]: do we also need a check for `if max_global_steps is not None and global_step == max_global_steps:` ?

File: submission_runner.py [hunk 2]
```diff
@@ -129,7 +129,7 @@
                      False,
                      'Whether to use Weights & Biases logging.')
 flags.DEFINE_boolean('profile', False, 'Whether to produce profiling output.')
-
+flags.DEFINE_integer('max_global_steps', None, 'Maximum number of update steps')
```
Comment (znado) [inline submission_runner.py:132]: nit: add `.` to end of help string.

**Review by znado — APPROVED** [review]
LGTM just minor fix

File: submission_runner.py (hunk 1 above)
Comment (priyakasimbeg) [reply submission_runner.py:304]: If max_global_steps is None then global_step == max_global_steps will be False. So I don't think we need to add an extra check.

**Pull request merged**